packages/swapper/src/swappers/RelaySwapper/getTradeQuote/getTradeQuote.ts

```diff
@@ -0,0 +1,242 @@
+import { makeSwapErrorRight } from '../../../utils'
+import type { relayChainMap as relayChainMapImplementation } from '../constant'
+import { getRelayEvmAssetAddress } from '../utils/getRelayEvmAssetAddress'
+import { relayService } from '../utils/relayService'
+import { relayTokenToAsset } from '../utils/relayTokenToAsset'
+import { relayTokenToAssetId } from '../utils/relayTokenToAssetId'
+import type { QuoteParams, RelayTradeQuote, RelayTradeRate } from '../utils/types'
+import { isRelayToken } from '../utils/types'
+
+// @TODO: implement affiliate fees
+export const getQuote = async (
+  params: QuoteParams,
+  config: SwapperConfig,
+): Promise<Result<AxiosResponse<Execute, any>, SwapErrorRight>> => {
+  return await relayService.post<Execute>(`${config.VITE_RELAY_API_URL}/quote`, params)
+}
+
+export async function getTrade({
+  input,
+  deps,
+  relayChainMap,
+}: {
+  input: CommonTradeQuoteInput | GetTradeRateInput
+  deps: SwapperDeps
+  relayChainMap: typeof relayChainMapImplementation
+}): Promise<Result<RelayTradeQuote[] | RelayTradeRate[], SwapErrorRight>> {
+  const {
+    sellAsset,
+    buyAsset,
+    sellAmountIncludingProtocolFeesCryptoBaseUnit,
+    sendAddress,
+    receiveAddress,
+    accountNumber,
+    affiliateBps,
+    potentialAffiliateBps,
+    slippageTolerancePercentageDecimal: _slippageTolerancePercentageDecimal,
+  } = input
+
+  const slippageToleranceBps = _slippageTolerancePercentageDecimal
+    ? convertDecimalPercentageToBasisPoints(_slippageTolerancePercentageDecimal).toFixed()
+    : undefined
+
+  const sellRelayChainId = relayChainMap[sellAsset.chainId]
+  const buyRelayChainId = relayChainMap[buyAsset.chainId]
+
+  if (sellRelayChainId === undefined) {
+    return Err(
+      makeSwapErrorRight({
+        message: `asset '${sellAsset.name}' on chainId '${sellAsset.chainId}' not supported`,
+        code: TradeQuoteError.UnsupportedTradePair,
+      }),
+    )
+  }
+
+  if (buyRelayChainId === undefined) {
+    return Err(
+      makeSwapErrorRight({
+        message: `asset '${buyAsset.name}' on chainId '${buyAsset.chainId}' not supported`,
+        code: TradeQuoteError.UnsupportedTradePair,
+      }),
+    )
+  }
+
+  if (!sendAddress) {
+    return Err(
+      makeSwapErrorRight({
+        message: 'sendAddress is required',
+      }),
+    )
+  }
+
+  const maybeQuote = await getQuote(
+    {
+      originChainId: sellRelayChainId,
+      originCurrency: getRelayEvmAssetAddress(sellAsset),
+      destinationChainId: buyRelayChainId,
+      destinationCurrency: getRelayEvmAssetAddress(buyAsset),
+      tradeType: 'EXACT_INPUT',
+      amount: sellAmountIncludingProtocolFeesCryptoBaseUnit,
+      recipient: receiveAddress,
+      user: sendAddress,
+      slippageTolerance: slippageToleranceBps,
+    },
+    deps.config,
+  )
+
+  if (maybeQuote.isErr()) return Err(maybeQuote.unwrapErr())
+
+  const quote = maybeQuote.unwrap()
```

Comment:
nitpick: one last accessor to deal with

```suggestion
  const { data: quote } = maybeQuote.unwrap()
```